projects/2DoBox-Pivot.md

```diff
@@ -0,0 +1,160 @@
+
+# Phase 1
+
+### Adding a new ToDo
+
+On the application’s main page, a user should:
+
+See two text boxes for entering the “Title” and “Task” for a new TODO, and a “Save” button for committing that TODO.
+When a user clicks “Save”:
+
+A new TODO with the provided title and body should appear in the TODO list.
+The text fields should be cleared and ready to accept a new TODO.
+The page should not reload.
+The TODO should be persisted. It should still be present upon reloading the page
+
+
+### Deleting an existing TODO
+
+When viewing the TODO list:
+
+Each TODO in the list should have a link or button to “Delete” (or 𝗫).
+Upon clicking “Delete”, the appropriate TODO should be removed from the list.
+The page should not reload when an idea is deleted.
+The TODO should be removed from localStorage. It should not re-appear on next page load.
+
+
+### Editing an existing TODO
+
+When a user clicks the title or task of a TODO in the list, that text should become an editable text field, pre-populated with the existing  TODO title or task.
+The user should be able to “commit” their changes by pressing “Enter/Return” or by clicking outside of the text field.
+If the user reloads the page, their edits will be reflected.
+
+
+# Phase 2
+
+### Accessibility
+
+Your web application should pass `aXe-core` tests and should be `tab index` accessible.
+
+### Importance
+
+Each  TODO should be given a level of importance. As a user I should be able to change the level of importance by up-voting or down-voting that specific  TODO. Each  TODO should start with a level of `Normal`.
+
+Levels of Importance are as follows
+
+1) Critical
+
+2) High
+
+3) Normal
+
+4) Low
+
+5) None
+
+The change of importance should persist after a page refresh
+
+### Bookmarking
+
+There should be a `bookmark` button for each  TODO. As a user if I click the `bookmark` button I should see a visual change for the specific  TODO I've bookmarked.
+
+If I refresh the page the book mark should persist.
+
+# Phase 3
+
+### Recent TODOs
+
+The application should only show the ten most recent  TODOS.
+The application should contain a button labeled “Show more TODO S…” which will load additional messages from the past.
+
+### Filter by Importance
+
+The application should allow users to filter the TODO list based off of importance. Your application should have 5 buttons corresponding to each level of importance (Critical, High, Normal, Low, and None). When one of the filter buttons is clicked the TODO list should only display TODOs with the selected importance.
+
+### Character Counter
+
+The application is able to count the number of characters inside of the input field in real time.
+
+### Submit button disabled
+
+The submit button should be disabled when there is not valid content in both input fields and if the input field character count exceeds 120 characters.
+
+
```

Comment:
Are we doing any extensions? A neat extension might be to set a 'date due' - would be kind of a biggie tho...